test/subqueues/test.c

```diff
@@ -96,14 +101,17 @@ static int child_main(void)
   mtev_dso_post_init();
 
   mtev_conf_write_log();
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-function-type"
   mtev_conf_watch_and_journal_watchdog((int (*)(void *)) mtev_conf_write_log, NULL);
+#pragma GCC diagnostic pop
 
   eventer_jobq_t *test_jobq = eventer_jobq_create_ms("test_jobq", EVENTER_JOBQ_MS_GC);
   eventer_jobq_set_concurrency(test_jobq, 4);
 
-  for(int i=0; i<sizeof(work)/sizeof(*work); i++) {
+  for(uint32_t i=0; i<sizeof(work)/sizeof(*work); i++) {
```

Comment:
general question: when do we use `uint32_t` over `size_t`?

Reply:
For this change I was just trying to quash compiler warnings. In this case it was signed/unsigned mismatch and I was just trying to make the minimum change that worked. They could be changed to `size_t` if desired.

Reply:
This should be ``size_t`` for these.  Should stick to the same integer size

Reply:
OK will fix.